internal/logger/logger.go

```diff
@@ -39,29 +45,33 @@ func (l *Logger) Printf(format string, a ...any) {
 	l.printf(format, a...)
 }
 
-func (l *Logger) UpgradeStarting(guid string) {
+func (l *Logger) UpgradeStarting(name string, guid string) {
 	l.lock.Lock()
 	defer l.lock.Unlock()
 
-	l.printf("starting to upgrade %q", guid)
+	l.printf("starting to upgrade instance: %q guid: %q", name, guid)
 }
 
-func (l *Logger) UpgradeSucceeded(guid string, duration time.Duration) {
+func (l *Logger) UpgradeSucceeded(name string, guid string, duration time.Duration) {
 	l.lock.Lock()
 	defer l.lock.Unlock()
 
 	l.successes++
 	l.complete++
-	l.printf("finished upgrade of %q successfully after %s", guid, duration)
+	l.printf("finished upgrade of instance: %q guid: %q successfully after %s", name, guid, duration)
 }
 
-func (l *Logger) UpgradeFailed(guid string, duration time.Duration, err error) {
+func (l *Logger) UpgradeFailed(name string, guid string, duration time.Duration, err error) {
 	l.lock.Lock()
 	defer l.lock.Unlock()
 
-	l.failures[guid] = err
+	l.failures = append(l.failures, failure{
+		name: name,
+		id:   guid,
```

Comment:
I don't really mind whether we use `guid`, `uuid` or `id`, but I think it helps to be consistent. If you think `guid` is incorrect, then I'd be happy with a global rename (ideally in another PR)

Reply:
I don't think it was that intentional.`guid` makes lots of sense for the CLI